### ISSUE TITLE
component panel completely fails to appear, protoflux panel broken

### ISSUE THREAD

**Issue by yoshiyoshyosh** [BlueCyro/CherryPick#15]
with the new resonite update & 1.0.7 release, the component panel doesn't appear at all when pressing "add component", and the protoflux panel has a broken search bar. without mods or by disabling cherrypick with its settings, the component panel appears.

bug might be linux native only?

log file (using no mods other than cherrypick): https://gist.github.com/yoshiyoshyosh/9fc89feff801da478a5eba29798a7cc1

the exception alone, for brevity:
```
System.TypeInitializationException: The type initializer for 'CherryPick.CherryPicker' threw an exception. ---> System.MissingMethodException: void FrooxEngine.SettingPropertyAttribute..ctor(string,string,string,bool,long,string)
  at (wrapper managed-to-native) System.MonoCustomAttrs.GetCustomAttributesInternal(System.Reflection.ICustomAttributeProvider,System.Type,bool)
  at System.MonoCustomAttrs.GetCustomAttributesBase (System.Reflection.ICustomAttributeProvider obj, System.Type attributeType, System.Boolean inheritedOnly) [0x00013] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.MonoCustomAttrs.GetCustomAttributes (System.Reflection.ICustomAttributeProvider obj, System.Type attributeType, System.Boolean inherit) [0x00037] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Reflection.MonoField.GetCustomAttributes (System.Type attributeType, System.Boolean inherit) [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Attribute.GetCustomAttributes (System.Reflection.MemberInfo element, System.Type type, System.Boolean inherit) [0x00088] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Attribute.GetCustomAttribute (System.Reflection.MemberInfo element, System.Type attributeType, System.Boolean inherit) [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Attribute.GetCustomAttribute (System.Reflection.MemberInfo element, System.Type attributeType) [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Reflection.CustomAttributeExtensions.GetCustomAttribute (System.Reflection.MemberInfo element, System.Type attributeType) [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Reflection.CustomAttributeExtensions.GetCustomAttribute[T] (System.Reflection.MemberInfo element) [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at FrooxEngine.WorkerInitializer.Initialize (System.Type workerType) [0x0009a] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at (wrapper dynamic-method) CherryPick.WorkerInitializer_Reverse_Patches.CherryPick.WorkerInitializer_Reverse_Patches.GetInitInfo_Patch0(System.Type)
  at CherryPick.CherryPicker..cctor () [0x0003a] in <e938b20560d74d62afe479080bfe0de2>:0 
   --- End of inner exception stack trace ---
  at (wrapper managed-to-native) System.Object.__icall_wrapper_mono_generic_class_init(intptr)
  at CherryPick.ComponentSelector_Patcher.SetupUI_Prefix (FrooxEngine.ComponentSelector __instance, Elements.Core.LocaleString title, Elements.Core.float2 size, FrooxEngine.SyncRef`1[T] ____uiRoot, FrooxEngine.Sync`1[T] ____rootPath) [0x0039c] in <e938b20560d74d62afe479080bfe0de2>:0 
  at (wrapper dynamic-method) FrooxEngine.ComponentSelector.FrooxEngine.ComponentSelector.SetupUI_Patch1(FrooxEngine.ComponentSelector,Elements.Core.LocaleString,Elements.Core.float2)
  at FrooxEngine.ComponentSelector.SetupDefault () [0x00013] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at FrooxEngine.SceneInspector.OnAttachComponentPressed (FrooxEngine.IButton button, FrooxEngine.ButtonEventData eventData) [0x00045] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at FrooxEngine.UIX.Button.RunPressed (FrooxEngine.ButtonEventData eventData) [0x0003b] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at FrooxEngine.UIX.Button.OnPressBegin (FrooxEngine.UIX.Canvas+InteractionData eventData) [0x00035] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at FrooxEngine.UIX.InteractionElement.ProcessEvent (FrooxEngine.UIX.Canvas+InteractionData eventData) [0x00235] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at FrooxEngine.UIX.Canvas.ProcessTouchEvent (FrooxEngine.TouchEventInfo& eventInfo, System.Collections.Generic.List`1[T] filters) [0x00408] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at FrooxEngine.UIX.Canvas.OnTouch (FrooxEngine.TouchEventInfo& eventInfo) [0x00000] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at FrooxEngine.UIX.Canvas.FrooxEngine.ITouchable.OnTouch (FrooxEngine.TouchEventInfo& eventInfo) [0x00000] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
  at FrooxEngine.TouchSource.SendTouchEvent (FrooxEngine.ITouchable touchable, FrooxEngine.TouchEventInfo& touchInfo) [0x00018] in <fef63798ef7b4c9cb897f34ff12fb863>:0 
```

**Comment by BlueCyro** [comment]
At a glance, this appears to be voodoo magic. I'll keep looking into this, but as of right now I consider Linux native to be on the far reaches of livable society in terms of playing Resonite. I can't replicate this on my windows machine, and I haven't heard anything from those I know who play on Proton either. I don't touch the settings stuff anywhere in this codebase. Also try on the latest version of Resonite, as it's been updated to `2024.5.1.68` since last night. I don't anticipate that'll make much of a difference, but it's good to be thorough.

If this is indeed a Linux-native-only thing though, then I don't know how much help I can realistically give. It's just so broken at this point that I almost wish it wasn't presented to the user for download.

**Comment by yoshiyoshyosh** [comment]
so this was fixed in 1.1.0 and I wanted to wait unitl the type update and this updated again and... it's working. dunno if it just needed a recompile the entire time or some other change fixed this but... it's working now! so I'll close the issue
